usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -109,15 +109,119 @@ def get_total_results(keyword, sub_index, retries=3):
+                  "filter": {
+                    "terms": {
+                      "award_category": sub_index
+                    }
+                  }
+                }
+              }
+            }
+    return query
+
+
+def category_query_for_contract_nulls(keyword, sub_index):
+    query = {"query": {
+                    "bool": {
+                        "must": [
+                            {"query_string": {"query": keyword}},
+                            {"match": {"pulled_from": "IDV"}}
+                        ]
+                        }
+                    }
+             }
+    return query
+
+
+def clean_sub_index(sub_index):
+    lookup = {"loans": ["loans"], "other": ["other"], "contracts":
+              ["contract"], "direct_payments": ["direct", "payment"], "grants": ["grant"], "insurance": ["insurance"]}
+    try:
+        return lookup[(sub_index)]
+    except KeyError:
+        logger.error('Unexpected Response')
+        return None
+
+
+def get_total_results_by_award_category(keyword, sub_index, retries=3):
+    index_name = '{}*'.format(TRANSACTIONS_INDEX_ROOT)
+    sub_index = clean_sub_index(sub_index)
+    if sub_index == ["contract"]:
+        contracts_count = 0
+        query = category_query_for_contract_nulls(keyword, sub_index)
+        response = es_client_query(index=index_name, body=query, retries=retries)
+        if response:
+            try:
+                contracts_count += response['hits']['total']
+            except KeyError:
+                logger.error('Unexpected Response')
+        else:
+            logger.error('No Response')
+
+        query = category_query_for_all(keyword, sub_index)
+        response = es_client_query(index=index_name, body=query, retries=retries)
+
+        if response:
+            try:
+                contracts_count += response['hits']['total']
+            except KeyError:
+                logger.error('Unexpected Response')
+        else:
+            logger.error('No Response')
+
+        return contracts_count
+    elif sub_index == ["other"]:
+        other_count = 0
+
+        query = category_query_for_all(keyword, sub_index)
+        response = es_client_query(index=index_name, body=query, retries=retries)
+        if response:
+            try:
+                other_count += response['hits']['total']
+            except KeyError:
+                logger.error('Unexpected Response')
+        else:
+            logger.error('No Response')
+
+        query = category_query_for_all(keyword, ["insurance"])
+        response = es_client_query(index=index_name, body=query, retries=retries)
+
+        if response:
+            try:
+                other_count += response['hits']['total']
+            except KeyError:
+                logger.error('Unexpected Response')
+        else:
+            logger.error('No Response')
+
+        return other_count
+
+    else:
+        query = category_query_for_all(keyword, sub_index)
```

Comment:
how does this behave when sub_index = ["direct", "payment"]?

Reply:
it returns direct payments. you could do the same by simply searching for "direct" or "payment" for the data that is in there, but was being thorough.